### ISSUE TITLE
Failed to mount component: template or render function not defined.

### ISSUE THREAD

**Issue by nataliawww** [vue-styleguidist/vue-styleguidist#210]
Using vue-styleguidist with Nuxt 2.0 and getting this warning
`Cannot parse [mycomponent.vue] TypeError: Cannot read property 'lang' of null`
followed by a fail to compile with this error:
`vue-loader was used without the corresponding plugin. Make sure to include VueLoaderPlugin in your webpack config`

I am using the sass-loader and use scss to style the components

**Comment by viljamis** [comment]
@nataliawojtkowska Sounds like you might be trying to use vue-loader without the appropriate plugin and that this isn’t related directly to Vue Styleguidist. Double check that you’re importing and including VueLoaderPlugin like this:

```js
const { VueLoaderPlugin } = require("vue-loader")
```

and then in exports:

```js
module.exports = {
  // other config, and then somewhere:
  plugins: [new VueLoaderPlugin()]
}
```

**Comment by nataliawojtkowska** [comment]
Thanks, but including the plugin in my config throws an error on `run dev`

**Comment by nataliawojtkowska** [comment]
Alright this issue is now resolved by configuring the vue-loader in webpackConfig's rules. The styleguide compiles with no errors but the component won't load

```
[Vue warn]: Failed to mount component: template or render function not defined.

found in

---> <Logo> at components/Logo.vue
```

**Comment by elevatebart** [comment]
Hello @nataliawww it seems like you are trying to load Vue in a non standard way.

I had this issue a couple of times. Would you mind sharing your the contents of your `logo.vue` and the contents of `styleguide.config.js` ?

**Comment by elevatebart** [comment]
This error happens to me when I use the old way of starting up a Vue app. When in `main.js`, I use 
```js
new Vue({
  el: '#app',
  components: { App },
  template: '<App/>'
})
``` 
instead of 
```js
new Vue({
  render: h => h(App)
}).$mount('#app')
```

**Comment by Joulss** [comment]
Hello,
I'm facing the same issue.
First I just installed vue-stleguidist using "vue add styleguidist"

- First try : ran `yarn vue-styleguidist server`. Compilation OK

Result : blank page and `process is not defined` in the console

- Second try : add the following webpack config in styleguide.config.js :
```
webpackConfig: {
        module: {
          rules: [
            {
              test: /\.vue$/,
              exclude: /node_modules/,
              loader: "vue-loader"
            },
            {
              test: /\.js?$/,
              exclude: /node_modules/,
              loader: "babel-loader"
            },
            {
              test: /\.css$/,
              loader: "style-loader!css-loader"
            }
          ]
        },
      },
```
Compilation OK, result : The components appears in the web page, but no preview and `Failed to mount component: template or render function not defined` message in the console, for each component.

- Third try : add vue-loader to the styleguide.config.js :
```
const { VueLoaderPlugin } = require("vue-loader");

module.exports = {
    title: 'Default Style Guide',
    components: 'src/components/**/[A-Z]*.vue',
    webpackConfig: {
        module: {
          rules: [
            {
              test: /\.vue$/,
              exclude: /node_modules/,
              loader: "vue-loader"
            },
            // For js or css files:
            {
              test: /\.js?$/,
              exclude: /node_modules/,
              loader: "babel-loader"
            },
            {
              test: /\.css$/,
              loader: "style-loader!css-loader"
            }
          ]
        },
        plugins: [new VueLoaderPlugin()],
      },
    defaultExample: true,
}
```
Compilation fails, msg : `Error: [VueLoaderPlugin Error] No matching use for vue-loader is found.
Make sure the rule matching .vue files include vue-loader in its use.`

I don't know what to try next... Maybe you can help me. Thanks.

**Comment by Joulss** [comment]
For more information, here my main.js :

```
import Vue    from 'vue';
import App    from './App.vue';
import router from './router';
import store  from './store';

new Vue({
  router,
  store,
  render: h => h(App)
}).$mount('#app');
```
and one of the vue components (I choosed the smallest one, but it triggers the error as the others) :

```
<template>
  <div class="dtpd-pill">
    <slot></slot>
  </div>
</template>

<script>
  export default {
    name: 'Pill'
  }
</script>

<style lang="scss" scoped>
  .dtpd-pill {
    font-size        : 11px;
    width            : 22px;
    height           : 22px;
    border-radius    : 50%;
    background-color : #ff7900;
    color            : #fff;
    margin-right     : 10px;
    text-align       : center;
    line-height      : 12px;
  }
</style>
```

**Comment by Joulss** [comment]
Ok, I installed a fresh vue cli project and added one by one my files and my packages.
It works without any webpack config. The problem appears when a vue refers to `process.env`

**Comment by elevatebart** [comment]
Hello @Joulss 

If you look inside your `package.json`, you will find 2 new entries in the script field: `styleguide` and  `styleguide:build. When using the vue cli 3 you are not supposed to call vue-styleguidist directly but through the `vue-cli-service`.

Once `vue add styleguidist` run, use simple `yarn styleguide`. It should work better.
If the new scripts can't be found (maybe they are cleaned by another process) just copy [the standard cli example](https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/examples/vuecli3/package.json)'s script in you package json and run `yarn styleguide`

Sorry for not responding earlier and I hope this helps.
Bart

**Comment by Joulss** [comment]
Hello @elevatebart

Thanks for your support.
I forgot to mention I already tried with the package.json scripts.
Same result with ```yarn styleguide``` (which run ```vue-cli-service styleguidist```) : blank page and ```process is not defined``` error message in the console.

**Comment by elevatebart** [comment]
Now we are getting somewhere. Thank you, `process` does not exist in the browser where components are rendered and showcased. It is specifically a node object, in what context do you use `process.env`? Is it in the `<script>` block of a `.vue` file? Is it in a wrapper object? 

You can use [define-plugin for webpack](https://webpack.js.org/plugins/define-plugin/) to make sure that process is defined somehow. I would suggest that you rather use the following simple method:
Instead of using `if(process.env === "dev")` use `if(process && process.env === 'dev')`.

I hope this helps.
Bart
